subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/ModifierAnnotationCategory.java

```diff
@@ -32,31 +32,35 @@
 import java.util.Map;
 
 public enum ModifierAnnotationCategory implements AnnotationCategory {
-    INCREMENTAL(ImmutableSet.of(
+    INCREMENTAL("incremental",
         Incremental.class,
         SkipWhenEmpty.class
-    )),
-    NORMALIZATION(ImmutableSet.of(
+    ),
+    NORMALIZATION("normalization",
         Classpath.class,
         CompileClasspath.class,
         PathSensitive.class
-    )),
-    OPTIONAL(ImmutableSet.of(
+    ),
+    OPTIONAL("optional",
         Optional.class
-    )),
-    IGNORE_EMPTY_DIRECTORIES(ImmutableSet.of(
+    ),
+    IGNORE_EMPTY_DIRECTORIES("ignore empty directories",
         IgnoreEmptyDirectories.class
-    ));
+    );
 
+    private final String displayName;
     private final ImmutableSet<Class<? extends Annotation>> annotations;
 
-    ModifierAnnotationCategory(ImmutableSet<Class<? extends Annotation>> annotations) {
-        this.annotations = annotations;
+    @SafeVarargs
+    @SuppressWarnings("varargs")
+    ModifierAnnotationCategory(String displayName, Class<? extends Annotation>... annotations) {
+        this.displayName = displayName;
+        this.annotations = ImmutableSet.copyOf(annotations);
     }
 
     @Override
     public String getDisplayName() {
-        return name().toLowerCase();
+        return displayName;
```

Comment:
Why change the display name here?

Reply:
Because we don't want to do a `toLowerCase()` every time the `Comparator` runs.

Reply:
The this looks like a premature optimization to me, especially since the sets we have a very small.

Reply:
The sets are small but we put/get from them a lot, and we'd be creating a new `String` every time.